src/golang/cmd/server/handler/connect_integration.go

```diff
@@ -663,16 +680,17 @@ func ValidatePrerequisites(
 		return http.StatusOK, nil
 	}
 
-	if svc == integration.Email {
-		emailIntegrations, err := integrationRepo.GetByServiceAndUser(ctx, svc, userID, DB)
+	// These integrations should be unique.
+	if svc == integration.Email || svc == integration.Slack {
+		integrations, err := integrationRepo.GetByServiceAndUser(ctx, svc, userID, DB)
 		if err != nil {
 			return http.StatusInternalServerError, errors.Wrap(err, "Unable to verify if email is connected.")
 		}
 
-		if len(emailIntegrations) > 0 {
+		if len(integrations) > 0 {
 			return http.StatusBadRequest, errors.Newf(
 				"You already have an email integration %s connected.",
-				emailIntegrations[0].Name,
+				integrations[0].Name,
```

Comment:
Change the error message to reflect that this can be a slack integration as well. Same thing for the error message right above this.